src/routes/profile.svelte

```diff
@@ -3,21 +3,39 @@
   import isoForURL from '@/utils/iso-for-url.js';
 
   export async function preload(page, session) {
+    let requestedEmail = page.query.user;
     let timelineFetchedUntil = new Date();
     timelineFetchedUntil.setMonth(timelineFetchedUntil.getMonth() - 3);
+    const { currentUser } = await getInitialData(this, session, new Map([
+      ['currentUser', '/account'],
+    ]));
+    if (currentUser == null) {
+      this.error(403, 'Profile');
+    }
+    const isMe = requestedEmail == currentUser.email;
+    const allowed = requestedEmail == null || isMe || currentUser.is_admin;
+    // non-admin requested a different profile
+    if (!allowed) {
+      this.redirect(302, `/profile`);
```

Comment:
I think a 403 would work better. This is an access violation, and is not something that will naturally occur on the website, so it makes sense to report an error instead of silently redirecting to the allowed resource

Reply:
Quoting your messages on Telegram:
> Lev Chelyadinov, [16.02.20 08:49]
> Like, if they ever get this url

> Lev Chelyadinov, [16.02.20 08:49]
> They should be thrown to their own profile

> Lev Chelyadinov, [16.02.20 08:49]
> Instead of 403

You argued that it should be a query param instead of a route one for exactly this reason, to simply redirect by removing the query in case it wasn't the user's own profile

Reply:
Hm. What was I thinking. Okay then, let's leave it as it is :D